cardano-crypto-class/src/Cardano/Crypto/DSIGN/SECP256k1.hs

```diff
@@ -0,0 +1,113 @@
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE ConstraintKinds #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# OPTIONS_GHC -Wno-orphans #-} -- need NoThunks for secp256k1-haskell types
+
+module Cardano.Crypto.DSIGN.SECP256k1 where
+
+import Cardano.Binary (FromCBOR (fromCBOR), ToCBOR (toCBOR, encodedSizeExpr))
+import Data.ByteString (ByteString)
+import Crypto.Random (getRandomBytes)
+import Cardano.Crypto.Seed (runMonadRandomWithSeed)
+import Data.Serialize (Serialize (get, put), runPut, runGet)
+import Data.Kind (Type)
+import GHC.Generics (Generic)
+import Control.DeepSeq (NFData)
+import qualified Crypto.Secp256k1 as SECP
+import NoThunks.Class (NoThunks)
+import Cardano.Crypto.DSIGN.Class (DSIGNAlgorithm (VerKeyDSIGN, SignKeyDSIGN, SigDSIGN,
```

Comment:
Consider formatting these as one per line - like this, they're harder to read, and small changes will lead to more diff noise.